### PR TITLE
feat(docker): build Docker images too

### PR DESCRIPTION
**Pull request by hazcod**
##### Description

This adds-in an updated Dockerfile and enables Docker image building & push to GitHub. :sparkles: 

##### Type of change

- [x] New feature :sparkles:

##### How Has This Been Tested?

- [x] `docker build -t kubeaudit .`

##### Checklist:

- [x] I have :tophat: my changes (A 🎩 specifically includes pulling down changes, setting them up, and manually testing the changed features and potential side effects to make sure nothing is broken)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage did not decrease
- [x] I have signed the appropriate [Contributor License Agreement](https://cla.shopify.com/)

### CODE REVIEW

**Comment by hazcod** [comment]
Perhaps nice to keep an eye on https://github.com/goreleaser/goreleaser/issues/1783 too

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,52 @@
...
+WORKDIR /go/src/app/
+
+# download and cache our dependencies
+VOLUME /go/pkg/mod
+COPY go.mod go.sum ./
+RUN go mod download
+
+# compile kubeaudit
+COPY . ./
+RUN go build -a -ldflags '-w -s -extldflags "-static"' -o /go/bin/kubeaudit ./cmd/ \
```
Comment (caarlos0) [inline Dockerfile:25]: goreleaser will already provide the pre-compiled binary here, you don't need to compile it again (unless you really want to).

e.g. https://github.com/goreleaser/goreleaser/blob/master/Dockerfile#L16

Comment (hazcod) [reply Dockerfile:25]: I guess it's a double edged knife; on one side you want to simplify things and integrate with goreleaser, on the other side you also want to provide a full-docker build chain for people not needing/wanting Go on their system.

Comment (caarlos0) [reply Dockerfile:25]: In some projects I create a `goreleaser.dockerfile` just for goreleaser... but yeah, I get it, totally your call :)

**Review by genevieveluyt — APPROVED** [review]
This looks great thank you! The blocker is that I need to figure out how to get access to our Dockerhub account

**Comment by genevieveluyt** [comment]
Looks like the release (via goreleaser) failed on the docker build step (specifically on `COPY go.mod go.sum ./`) 😞

I'm able to build the image just fine with `docker build -t kubeaudit .`

Any ideas @hazcod  @caarlos0 ?

```
   ⨯ release failed after 13.00s error=failed to build docker image:
Sending build context to Docker daemon  30.02MB
Step 1/18 : FROM golang:1.15.1 AS builder
 ---> 9f495162f677
Step 2/18 : ENV CGO_ENABLED=0 GOOS=linux GOARCH=amd64
 ---> Using cache
 ---> fc632417f30c
Step 3/18 : RUN apt-get install --yes --no-install-recommends ca-certificates tzdata
 ---> Using cache
 ---> aeb926f66ea0
Step 4/18 : RUN adduser --shell /bin/true --uid 1000 --disabled-login --no-create-home --gecos '' app   && sed -i -r "/^(app|root)/!d" /etc/group /etc/passwd   && sed -i -r 's#^(.*):[^:]*$#\1:/sbin/nologin#' /etc/passwd
 ---> Using cache
 ---> eabeaa51d512
Step 5/18 : WORKDIR /go/src/app/
 ---> Using cache
 ---> db516cc61ace
Step 6/18 : VOLUME /go/pkg/mod
 ---> Using cache
 ---> 50c3f49b62ed
Step 7/18 : COPY go.mod go.sum ./
COPY failed: stat /var/lib/docker/tmp/docker-builder113941890/go.mod: no such file or directory
: exit status 1
```

**Comment by genevieveluyt** [comment]
Ah ok thank you for clarifying, that worked and we are live! https://hub.docker.com/repository/docker/shopify/kubeaudit/

**Pull request merged**